contrib/localnet/orchestrator/smoketest/test_bitcoin.go

```diff
@@ -165,6 +168,87 @@ func (sm *SmokeTest) DepositBTC() {
 			break
 		}
 	}
+
+	// prove the two transactions of the deposit
+	LoudPrintf("Bitcoin Merkle Proof\n")
+
+	sm.ProveBTCTransaction(txHash_1)
+	sm.ProveBTCTransaction(txHash_2)
+}
+
+func (sm *SmokeTest) ProveBTCTransaction(txHash *chainhash.Hash) {
+	// get tx result
+	btc := sm.btcRPCClient
+	txResult, err := btc.GetTransaction(txHash)
+	if err != nil {
+		panic("should get outTx result")
+	}
+	if txResult.Confirmations <= 0 {
+		panic("outTx should have already confirmed")
+	}
+	txBytes, err := hex.DecodeString(txResult.Hex)
+	if err != nil {
+		panic(err)
+	}
+
+	// get the block with verbose transactions
+	blockHash, err := chainhash.NewHashFromStr(txResult.BlockHash)
+	if err != nil {
+		panic(err)
+	}
+	blockVerbose, err := btc.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		panic("should get block verbose tx")
+	}
+
+	// get the block header
+	header, err := btc.GetBlockHeader(blockHash)
+	if err != nil {
+		panic("should get block header")
+	}
+
+	// collect all the txs in the block
+	txns := []*btcutil.Tx{}
+	for _, res := range blockVerbose.Tx {
+		txBytes, err := hex.DecodeString(res.Hex)
+		if err != nil {
+			panic(err)
+		}
+		tx, err := btcutil.NewTxFromBytes(txBytes)
+		if err != nil {
+			panic(err)
+		}
+		txns = append(txns, tx)
+	}
+
+	// build merkle proof
+	mk := bitcoin.NewMerkle(txns)
+	path, index, err := mk.BuildMerkleProof(int(txResult.BlockIndex))
+	if err != nil {
+		panic("should build merkle proof")
+	}
+
+	// verify merkle proof statically
+	pass := bitcoin.Prove(*txHash, header.MerkleRoot, path, index)
+	if !pass {
+		panic("should verify merkle proof")
+	}
+
+	// verify merkle proof through RPC
+	res, err := sm.observerClient.Prove(context.Background(), &observertypes.QueryProveRequest{
+		ChainId:   common.BtcRegtestChain().ChainId,
+		TxHash:    txHash.String(),
+		BlockHash: blockHash.String(),
+		Proof:     common.NewBitcoinProof(txBytes, path, index),
+		TxIndex:   0, // bitcoin doesn't use txIndex
```

Comment:
>// bitcoin doesn't use txIndex

Maybe it's an attribute we can add in the generic `Proof` object then. Not needed for this PR

Reply:
That's a good point. Here, the `index` is the sibling index (flags) on the path to merkle root. It IS a general param in merkle proof but we can wait until it becomes general when Zetachain gets connected to more types of chains (more types of merkle trees involved).